src/relay/pass/pass_manager.cc

```diff
@@ -37,42 +37,46 @@ namespace transform {
 
 using tvm::IRPrinter;
 
-/*!
- * \brief A data structure to map the names of specific optimizations to
- *        numeric optimization levels
- */
-class OptPassLevel {
- public:
-  /*!
-   * \brief Get level for an optimization pass
-   *
-   * \param key pass name
-   * \return int level
-   */
-  int operator[](const std::string& key) const {
-    const auto data = CreateMap();
-    auto it = data.find(key);
-    if (it == data.end()) {
-      return -1;
-    }
-    return it->second;
+namespace {
+
+// TODO(zhiics) Maybe we can use PackedFunc here so that parameters can be
+// handled because we need to register the pass for Python invocation anyway.
+Pass GetPass(const std::string& pass_name) {
+  if (pass_name == "infer_type") {
+    return InferType();
+  } else if (pass_name == "alter_op_layout") {
+    return AlterOpLayout();
+  } else if (pass_name == "canonicalize_ops") {
+    return CanonicalizeOps();
+  } else if (pass_name == "combine_parallel_conv2d") {
+    return CombineParallelConv2D();
+  } else if (pass_name == "dead_code_elimination") {
+    return DeadCodeElimination();
+  } else if (pass_name == "eliminate_common_subexpr") {
+    return DeadCodeElimination();
+  } else if (pass_name == "fold_constant") {
+    return FoldConstant();
+  } else if (pass_name == "backward_fold_scale_axis") {
+    return FoldScaleAxis();
+  } else if (pass_name == "forward_fold_scale_axis") {
+    return FoldScaleAxis();
+  } else if (pass_name == "fold_scale_axis") {
+    return FoldScaleAxis();
+  } else if (pass_name == "partial_eval") {
+    return SimplifyInference();
+  } else if (pass_name == "simplify_inference") {
+    return SimplifyInference();
+  } else if (pass_name == "to_a_normal_form") {
+    return ToANormalForm();
+  } else if (pass_name == "to_graph_normal_form") {
+    return ToGraphNormalForm();
+  } else {
+    LOG(WARNING) << pass_name << " has not been registered yet." << "\n";
```

Comment:
LOG(FATAL)

Reply:
I think we may just want to give a warning because users may provide their own pass from python. We cannot know the pass name but we still should execute them with a warning.

Reply:
Depends on the user of GetPass, if it is only used to resolve dependencies, we should throw an error, we can, of course, implement it in a different way so that python side pass work as well later.

Reply:
@tqchen I agree that we should throw an error if it is used to resolve dependency. But currently we don't distinguish them. I would prefer to keep the warning for now and change it to error. Otherwise, I can disable the customized python tests for now and enable them later when we have different ways to execute them. How do you think?

Reply:
I am not sure how python side test would relate to GetPass, I thought it is only used by Sequential to resolve dependencies(as users can use transform.PassName() to construct the pass pragramatically)

Reply:
yeah, I wrote a few python passes and put in the list as a `Sequential`. We needed to check if a pass is enabled so that we needed to GetPass. But anyway, I think its more appropriate to put Python passes as `required`. Now I changed it to fatal.